src/store/connection/actions.js

```diff
@@ -76,3 +100,38 @@ export const getConnectionsCount = () => async (dispatch, getState) => {
     dispatch(fetchFailure());
   }
 };
+
+export const newConnectionAction = () => async (dispatch, getState) => {
+  const {
+    connections: {
+      host,
+      port,
+      nameDB,
+      username: name,
+      password,
+      type,
+      nameConnection,
+    }
+  } = getState();
+
+  if (!name || !password || !host || !port || !nameDB || !type || !nameConnection) {
+    return dispatch(emptyFieldsError());
+  }
+  dispatch(createConnectionStart());
+  try {
+    const data = await post(ADMIN_CONNECTIONS_CREATE_ENDPOINT, {
+      data: {
+        host,
+        port,
+        name: nameConnection,
+        databaseName: nameDB,
+        username: name,
+        password,
+        typeId: Number(type),
+      }
+    });
+    dispatch(setConnections(data));
+  } catch (e) {
+    dispatch(createConnectionFailure());
+  }
+};
```

Comment:
How do you handle errors ?

Reply:
There is no effect on the popup, apart from demonstrating error text at the bottom